### PR TITLE
__init__.py: Import classes arranged Alphabetical

### PR DESCRIPTION
**Pull request by Mah1ndra**
Arranged Imported classes in Alphabetical order

Closes: https://github.com/coala/coala/issues/5745

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [x] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `corobo mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

**Comment by Mah1ndra** [comment]
I followed the guidelines and made commits accordingly. When I ran coala locally I've got a warning but no errors. 
image

**Comment by mach64** [comment]
You have to ensure all tests pass locally before you push it to github. This includes not having any warnings thrown by pytest. Follow [this](https://api.coala.io/en/latest/Developers/Newcomers_Guide.html#step-3-creating-a-fork-and-testing-your-changes) to properly test your changes.

**Comment by mach64** [comment]
The CI builds are failing because your local repo is out of sync with master. You'll have to [rebase](https://coala.io/rebase) first.

**Comment by mach64** [comment]
Nice work with the testing. The commit message still seems a bit out of place. How about "__init__.py: Arrange imports alphabetically"? It is short and crisp, and still conveys the relevant information about the commit.

**Comment by pareksha** [comment]
I agree with @mach64, see if you can add language word in the commit msg too, something like `__init__: Arrange lang imports alphabetically`, as `__init__` doesn't tell much about the file.

File: coalib/bearlib/languages/__init__.py [hunk 1]
```diff
@@ -3,40 +3,56 @@
...
-from .definitions.Shell import Shell
-from .definitions.Jinja2 import Jinja2
+
+from .definitions.Unknown import Unknown
+
+from .definitions.Vala import Vala
 from .definitions.VisualBasic import VisualBasic
+
 from .definitions.XML import XML
+
```
Comment (pareksha) [inline coalib/bearlib/languages/__init__.py]: I see you are adding a lot of blank lines. IMO these are not required, please only arrange imports in alphabetic order, don't add/remove lines unnecessarily.

Comment (Mah1ndra) [reply coalib/bearlib/languages/__init__.py]: okay. Thank you for the review @pareksha  @mach64

Comment (Mah1ndra) [reply coalib/bearlib/languages/__init__.py]: i'll make the necessary changes as suggested

File: coalib/bearlib/languages/__init__.py [hunk 2]
```diff
@@ -3,39 +3,38 @@
...
 from .definitions.CSS import CSS
+from .definitions.CSharp import CSharp
 from .definitions.Fortran import Fortran
 from .definitions.Golang import Golang
 from .definitions.html import HTML
-from .definitions.Java import Java
-from .definitions.JavaScript import JavaScript
 from .definitions.JSON import JSON
 from .definitions.JSP import JSP
+from .definitions.Java import Java
```
Comment (shikharvaish28) [inline coalib/bearlib/languages/__init__.py]: @Mah1ndra ```Java``` should be above ```JSON``` alphabetically. (Are you considering Upper-case to be higher than Lower-case?)

File: coalib/bearlib/languages/__init__.py [hunk 3]
```diff
@@ -3,39 +3,38 @@
...
+from .definitions.JavaScript import JavaScript
```
Comment (shikharvaish28) [inline coalib/bearlib/languages/__init__.py]: (Same as the above comment)

**Comment by shikharvaish28** [comment]
I think python 3.4 test generally fails. (Though you can confirm with others too)

File: coalib/bearlib/languages/__init__.py (hunk 3 above)
Comment (Mah1ndra) [reply coalib/bearlib/languages/__init__.py]: Alright. I'll resolve these.

**Comment by Mah1ndra** [comment]
Do you mean which lines I shouldn't move? There are only two .Language imports and rest are .definition imports

**Comment by shashwatmishra76** [comment]
Yeah.Do not move the .Language imports.You should keep the imports from two different files i.e. from Language and definitions SEPARATED so as to promote better readability of the code.

**Comment by Mah1ndra** [comment]
i've already tried it. using lexicographically exceeding char limit

**Comment by Mah1ndra** [comment]
Removing lang and replacing lexico caused me pytest error @Monal5031

**Comment by Monal5031** [comment]
@Mah1ndra Read my message again. I asked to remove lang part and replace alpha with lexicographically. and since we are removing lang from head you will have to explain it in shortlog.
Does that make it clear?

**Comment by Monal5031** [comment]
And from what I can see it is due to something else the tests are failing, (how would a commit message make your tests fail? :confused: )

**Comment by Mah1ndra** [comment]
okay sure @atb00ker Thank you for your review :smile:

**Comment by atb00ker** [comment]
I read the logs for this commit as well, the ci/circleci: python-3.4 seems to have failed for reasons other than the changes in this PR but please wait for review from another member just to be sure! :smile:

**Comment by Mah1ndra** [comment]
Thank you for your review @atb00ker :smiley:

File: coalib/bearlib/languages/__init__.py [hunk 4]
```diff
@@ -3,40 +3,56 @@
...
-from .Language import Languages
-
-from .definitions.Unknown import Unknown
 from .definitions.Bash import Bash
+
 from .definitions.C import C
 from .definitions.CPP import CPP
 from .definitions.CSharp import CSharp
 from .definitions.CSS import CSS
+
```
Comment (shikharvaish28) [inline coalib/bearlib/languages/__init__.py]: I don't think there is any need for spaces like these. You may want to remove them.

File: coalib/bearlib/languages/__init__.py [hunk 5]
```diff
@@ -3,40 +3,56 @@
...
-from .definitions.m4 import m4
-from .definitions.Matlab import Matlab
+
+from .Language import Language
+from .Language import Languages
+
 from .definitions.Markdown import Markdown
+from .definitions.Matlab import Matlab
+from .definitions.m4 import m4
+
```
Comment (shikharvaish28) [inline coalib/bearlib/languages/__init__.py]: There are a lot of blank spaces which IMO are not needed.

File: coalib/bearlib/languages/__init__.py (hunk 1 above)
Comment (AkshJain99) [reply coalib/bearlib/languages/__init__.py]: As suggested by others remove unnecessary blank spaces from the code

**Comment by AkshJain99** [comment]
First of all as per coala guidelines of handling pull request there is only one commit per pull request ,so you should squash all the commits into single commit .
secondly try to understand gitmate changes ,your commit message is not as per gitmate bot.
refer this https://api.coala.io/en/latest/Developers/Writing_Good_Commits.html
and http://api.coala.io/en/latest/Developers/Git_Basics.html

**Comment by jayvdb** [comment]
Abandoned messy PR.

**Pull request closed without merging**